consensus/propagation/reactor.go

```diff
@@ -311,15 +323,15 @@ func ConcurrentRequestLimit(peersCount, partsCount int) int64 {
 // getPeer returns the peer state for the given peer. If the peer does not exist,
 // nil is returned.
 func (blockProp *Reactor) getPeer(peer p2p.ID) *PeerState {
-	blockProp.mtx.RLock()
-	defer blockProp.mtx.RUnlock()
+	blockProp.mtx.Lock()
+	defer blockProp.mtx.Unlock()
```

Comment:
unrelated, but apparently using a read/write mutex is a premature optimization unless reads take a long time, and the normal mutex is 10x simpler